ElectronClient/gui/KeymapConfig/ShortcutRecorder.tsx

```diff
@@ -0,0 +1,119 @@
+import * as React from 'react';
+import { useState, KeyboardEvent } from 'react';
+
+import styles_ from './styles';
+import { KeymapConfigScreenProps } from './KeymapConfigScreen';
+const { shim } = require('lib/shim');
+const { _ } = require('lib/locale.js');
+
+export interface ShortcutRecorderProps extends KeymapConfigScreenProps {
+    updateAccelerator: (accelerator: string) => void,
+    updateIsEditing: (isEditing: boolean) => void,
+}
+
+export const ShortcutRecorder = (props: ShortcutRecorderProps) => {
+	const styles = styles_(props);
+	const [accelerator, setAccelerator] = useState('');
+
+	const handleKeydown = (event: KeyboardEvent<HTMLDivElement>) => {
+		event.preventDefault();
+		const newAccelerator = KeymapService_domToElectronAccelerator(event);
+
+		switch (newAccelerator) {
+		case 'Enter':
+			props.updateAccelerator(accelerator);
+			props.updateIsEditing(false);
+			break;
+		case 'Escape':
+			return props.updateIsEditing(false);
+		default:
```

Comment:
A child component should not directly set data on the parent. Instead please use event handlers: `onSubmit`, `onCancel`. All React components should be controlled ones: https://reactjs.org/docs/forms.html#controlled-components